### ISSUE TITLE
As a developer, I want to add new columns or change the data structure without breaking everything for persistent RAMS

### ISSUE THREAD

**Issue by kitsuta** [magfest/ubersystem#2325]

**Comment by RobRuana** [comment]
Solved with alembic migrations, see #199.

**Comment by kitsuta** [comment]
So we'd be changing old data for migrations if it doesn't work, then?

**Comment by RobRuana** [comment]
I'm not sure what you mean

**Comment by kitsuta** [comment]
Historically, we make a ton of 'breaking' changes every year. If we have data from five years ago, and we decide the relevant column needs to change from a string to a choice opt, are we going to massage the old data to fit?

**Comment by RobRuana** [comment]
For all systems from this point forward we can apply migrations to upgrade their database. For all historic systems, we'll have to massage the data. No way around that in any case.

**Comment by kitsuta** [comment]
That's not an inevitability. We'd only need to do that if we used one instance for all our current data and our old data, which is the architecture you proposed. Using a manager app that collects the data means we'd keep the old data the same.

**Comment by RobRuana** [comment]
The manager app would still have to reconcile the data types, or at least understand that the data it is collecting will be different from one system to the next.

Somewhere, at some point, some process will have to massage the old data.

**Comment by kitsuta** [comment]
Welllllll I was thinking of 'massage' as 'directly change' but yeah, we do have to DEAL with the old data types in some way.

**Comment by RobRuana** [comment]
I guess we should leave this open to indicate that we need to deal with that historic data at some point.

**Comment by kitsuta** [comment]
Yeah, that was my thinking. It's part of the Persistent Epic, I wasn't thinking we'd close any of those issues before we made at least the first pass of the persistent RAMS.

**Comment by RobRuana** [comment]
The cool thing is that all of our current databases are versioned now. So we can easily upgrade them, or even keep them upgraded by deploying to them on a regular basis.
